cpp/include/cuvs/neighbors/cagra.hpp

```diff
@@ -104,6 +105,16 @@ struct index_params : cuvs::neighbors::index_params {
                graph_build_params::ivf_pq_params,
                graph_build_params::nn_descent_params>
     graph_build_params;
+  /**
+   * Whether to add the dataset content to the index, i.e.:
+   *
+   *  - `true` means the index is filled with the dataset vectors and ready to search after calling
+   * `build` provided there is enough memory available.
+   *  - `false` means `build` only builds the graph and the user is expected to
+   * update the dataset using cuvs::neighbors::cagra::update_dataset. CAGRA does not have `extent`
+   * API.
+   */
+  bool populate_data = true;
```

Comment:
I think we should use a more descriptive naming here- this doesn't say much at all. What does it mean to populate data? I'd take that to mean we have an array and we're filling it up with something, but I believe what we are really doing is attaching a dataset to the index, right? 

I propose we name this `attach_dataset_on_build` so there's no confusion here. That's also more inline with the naming used in the IVF methods: `add_data_on_build`.